includes/tn_upload.form.inc

```diff
@@ -59,7 +59,15 @@ function islandora_entities_get_tn_upload_form($form_state = NULL) {
  *   The drupal form state.
  */
 function islandora_entities_tn_upload_form_submit(array $form, array &$form_state) {
-  $object = $form_state['islandora']['objects'][0];
+  // Is this a thumbnail update?
+  $factory_object = islandora_object_load($form_state['islandora']['objects'][0]->id);
```

Comment:
Please use this wrapper function instead:

```
$factory_object = islandora_ingest_form_get_object($form_state);
```

https://github.com/Islandora/islandora/blob/9ac1eed4b9757345fc5848b6fd1695e0a2cd474d/includes/ingest.form.inc#L805-L824